general/array.hpp

```diff
@@ -92,8 +93,8 @@ class Array
    inline Array(const Array<CT> &src);
 
    /// Deep copy from a braced init-list of convertible type
-   template <typename CT, int N>
-   explicit inline Array(const CT (&values)[N]);
+   template <typename CT>
+   explicit inline Array(std::initializer_list<CT> values);
```

Comment:
I think we should keep the original implementation -- it allows construction which is not allowed by the new version.

Reply:
You mean initializing the Array with a C-style array with static size? I re-added the previous one (in addition to the new one) in c7bb0201c0745a17c4a37874315a91a942e49a6f.